src/polycap-description.c

```diff
@@ -0,0 +1,455 @@
+	description->n_cap = n_cap;
+	description->d_source = d_source;
+//	description->d_screen = d_screen;
+	description->src_x = src_x;
+	description->src_y = src_y;
+	description->src_sigx = src_sigx;
+	description->src_sigy = src_sigy;
+	description->src_shiftx = src_shiftx;
+	description->src_shifty = src_shifty;
+	description->nelem = nelem;
+	description->density = density;
+	for(i=0; i<description->nelem; i++){
+		description->iz[i] = iz[i];
+		description->wi[i] = wi[i]; //assumes weights are already provided as fractions (not percentages)
+	}
+
+	// Check whether weights add to 1
+	polycap_description_check_weight(description->nelem, description->wi);
+
+	//allocate profile memory
+	description->profile = malloc(sizeof(polycap_profile));
+	if(description->profile == NULL){
+		printf("Could not allocate description->profile memory.\n");
+		exit(1);
+	}
+	description->profile->z = malloc(sizeof(double)*(profile->nmax+1));
+	if(description->profile->z == NULL){
+		printf("Could not allocate description->profile->z memory.\n");
+		exit(1);
+	}
+	description->profile->cap = malloc(sizeof(double)*(profile->nmax+1));
+	if(description->profile->cap == NULL){
+		printf("Could not allocate description->profile->cap memory.\n");
+		exit(1);
+	}
+	description->profile->ext = malloc(sizeof(double)*(profile->nmax+1));
+	if(description->profile->ext == NULL){
+		printf("Could not allocate description->profile->ext memory.\n");
+		exit(1);
+	}
+	
+	// copy description->profile values to profile
+	for(i=0;i<=profile->nmax;i++){
+		description->profile->z[i] = profile->z[i];
+		description->profile->cap[i] = profile->cap[i];
+		description->profile->ext[i] = profile->ext[i];
+	}
+
+	// Calculate open area
+	description->open_area = (description->profile->cap[0]/description->profile->ext[0]) * (description->profile->cap[0]/description->profile->ext[0]) * description->n_cap;
+
+	return description;
+}
+
+//===========================================
+// get the polycap_profile from a polycap_description
+polycap_profile* polycap_description_get_profile(polycap_description *description)
+{
+	polycap_profile *profile;
+	int i;
+
+	//allocate profile memory
+	profile = malloc(sizeof(polycap_profile));
+	if(profile == NULL){
+		printf("Could not allocate profile memory.\n");
+		exit(1);
+	}
+	profile->nmax = description->profile->nmax;
+	profile->z = malloc(sizeof(double)*(profile->nmax+1));
+	if(profile->z == NULL){
+		printf("Could not allocate profile->z memory.\n");
+		exit(1);
+	}
+	profile->cap = malloc(sizeof(double)*(profile->nmax+1));
+	if(profile->cap == NULL){
+		printf("Could not allocate profile->cap memory.\n");
+		exit(1);
+	}
+	profile->ext = malloc(sizeof(double)*(profile->nmax+1));
+	if(profile->ext == NULL){
+		printf("Could not allocate profile->ext memory.\n");
+		exit(1);
+	}
+
+	// copy description->profile values to profile
+	for(i=0;i<=profile->nmax;i++){
+		profile->z[i] = description->profile->z[i];
```

Comment:
Is het niet beter om altijd gewoon `z`, `cap` en `ext`uit `description->profile` te gebruiken ipv te werken met een exacte kopie ervan in `profile`?

Indien zo, verwijder gewoon die 3 member variables uit de definitie van `polycap_profile`

Reply:
Ben hier niet echt mee wat je bedoelt. Lijkt me sterk gelinked aan de opmerking hierboven, dus misschien is het 'probleem' hierboven reeds opgelost?

Reply:
Je hebt hierboven al in `polycap_description_new`, een kopie gemaakt van `profile` en toegewezen aan `description->profile`: je kan er dus zeker van zijn dat de gebruiker dus niet meer kan foefelen met wat daarin zit. Het is daarom zinloos volgens mij om nog eens kopie te maken van wat er in `profile` zit en dat dan te gaan steken in `description`. Best gewoon te werken met `description->profile`